File: algorithms/Codility-master/Codility-master/3_TimeComplexity/TapeEquilibrium.js

// 100%
function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    if (A.length <=1){
        return 0;
    }
    if (A.length == 2){
        return Math.abs(A[1] - A[0]);
    }
    
    let sum_l = A[0];
    let sum_r = A.reduce((a, b) => a + b) - A[0]
    let dis = Math.abs(sum_l - sum_r);
    
    for (let i=1; i < (A.length - 1); i++){
        sum_l += A[i];
        sum_r -= A[i];
        dis = Math.min(dis, Math.abs(sum_l - sum_r));
    }
    return dis;   
}

File: algorithms/Codility-master/Codility-master/3_TimeComplexity/TapeEquilibrium.py


# 100%
def solution(A):
    # write your code in Python 3.6
    if len(A) <= 1:
        return 0
    if len(A) == 2:
        return abs(A[1] - A[0])
        
    sum_l = A[0]
    sum_r = sum(A[1:])
    dis = abs(sum_l - sum_r)
    for i in range(1, len(A)-1):
        sum_l += A[i]
        sum_r -= A[i]
        dis = min(dis, abs(sum_l - sum_r))
        
    return dis
        